multiversx_sdk/network_providers/api_network_provider.py

```diff
@@ -288,7 +291,19 @@ def do_post_generic(self, url: str, data: Any, url_parameters: Optional[dict[str
 
     def _do_get(self, url: str) -> Any:
         try:
-            response = requests.get(url, **self.config.requests_options)
+            retry_strategy = Retry(
+                total=self.config.requests_options.get("retries"),
+                backoff_factor=self.config.requests_options.get("backoff_factor", 0),
+                status_forcelist=self.config.requests_options.get("status_forcelist"),
+            )
+
+            adapter = HTTPAdapter(max_retries=retry_strategy)
+
+            with requests.Session() as session:
+                session.mount("https://", adapter)
```

Comment:
Might not be reliable at times. E.g. when the developer is using a local proxy.

Reply:
also added:
```
session.mount("http://", adapter)
```